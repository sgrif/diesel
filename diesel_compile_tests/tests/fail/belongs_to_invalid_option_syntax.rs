#[macro_use]
extern crate diesel;

table! {
    bar (id) {
        id -> Integer,
    }
}

table! {
    foo (bar_id) {
        bar_id -> Integer,
    }
}

#[derive(Identifiable)]
#[diesel(table_name = bar)]
struct Bar {
    id: i32,
}

#[derive(Identifiable)]
#[diesel(table_name = bar)]
struct Baz {
    id: i32,
}

#[derive(Associations)]
#[diesel(belongs_to)]
#[diesel(belongs_to = "Bar")]
#[diesel(belongs_to())]
#[diesel(belongs_to(foreign_key = bar_id))]
#[diesel(belongs_to(Bar, foreign_key))]
#[diesel(belongs_to(Bar, foreign_key(bar_id)))]
#[diesel(belongs_to(Baz, foreign_key = bar_id, random_option))]
#[diesel(table_name = foo)]
struct Foo {
    bar_id: i32,
}

fn main() {}
