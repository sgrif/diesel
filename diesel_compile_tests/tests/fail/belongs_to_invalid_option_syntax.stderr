error: `belongs_to` is either unexpected or has bad format
  --> $DIR/belongs_to_invalid_option_syntax.rs:29:10
   |
29 | #[diesel(belongs_to)]
   |          ^^^^^^^^^^
